packages/cli/src/commands/deployment/deploy.ts

```diff
@@ -15,59 +15,26 @@ import {
 import {addV, checkToken, promptWithDefaultValues, valueOrPrompt} from '../../utils';
 
 export default class Deploy extends Command {
   static description = 'Deployment to hosted service';
 
-  static flags = {
-    org: Flags.string({description: 'Enter organization name'}),
-    projectName: Flags.string({description: 'Enter project name'}),
+  static flags = Object.assign(defaultCommandFlags(), {
     ipfsCID: Flags.string({description: 'Enter IPFS CID'}),
-
-    type: Flags.string({options: ['stage', 'primary'], default: DEFAULT_DEPLOYMENT_TYPE, required: false}),
-    indexerVersion: Flags.string({description: 'Enter indexer-version', required: false}),
-    queryVersion: Flags.string({description: 'Enter query-version', required: false}),
-    dict: Flags.string({description: 'Enter dictionary', required: false}),
     endpoint: Flags.string({description: 'Enter endpoint', required: true}),
-    //indexer set up flags
-    indexerUnsafe: Flags.boolean({description: 'Enable indexer unsafe', required: false}),
-    indexerBatchSize: Flags.integer({description: 'Enter batchSize from 1 to 30', required: false}),
-    indexerSubscription: Flags.boolean({description: 'Enable Indexer subscription', required: false}),
-    disableHistorical: Flags.boolean({description: 'Disable Historical Data', required: false}),
-    indexerUnfinalized: Flags.boolean({
-      description: 'Index unfinalized blocks (requires Historical to be enabled)',
-      required: false,
-    }),
-    indexerStoreCacheThreshold: Flags.integer({
-      description: 'The number of items kept in the cache before flushing',
-      required: false,
-    }),
-    disableIndexerStoreCacheAsync: Flags.boolean({
-      description: 'If enabled the store cache will flush data asynchronously relative to indexing data.',
-      required: false,
-    }),
-    indexerWorkers: Flags.integer({description: 'Enter worker threads from 1 to 5', required: false, max: 5}),
-
-    //query flags
-    queryUnsafe: Flags.boolean({description: 'Enable indexer unsafe', required: false}),
-    querySubscription: Flags.boolean({description: 'Enable Query subscription', required: false}),
-    queryTimeout: Flags.integer({description: 'Enter timeout from 1000ms to 60000ms', required: false}),
-    queryMaxConnection: Flags.integer({description: 'Enter MaxConnection from 1 to 10', required: false}),
-    queryAggregate: Flags.boolean({description: 'Enable Aggregate', required: false}),
-
-    useDefaults: Flags.boolean({
-      char: 'd',
-      description: 'Use default values for indexerVersion, queryVersion, dictionary, endpoint',
-      required: false,
-    }),
-  };
+  });
 
   async run(): Promise<void> {
     const {flags} = await this.parse(Deploy);
-    let {dict, endpoint, indexerVersion, ipfsCID, org, projectName, queryVersion} = flags;
+    let _flags: DeploymentFlagsInterface = flags as unknown as DeploymentFlagsInterface;
```

Comment:
This cast is unsafe, the return type from parse should provide sufficient types already